src/server/hook_func.c

```diff
@@ -4608,10 +4612,20 @@ add_mom_hook_action(mom_hook_action_t ***hookact_array,
 						"add_mom_hook_action", log_buffer);
 					return (-1);
 				}
-				if (set_action)
+				if (set_action) {
 					pact->action = action;
-				else
+				} else {
+					if ((action & MOM_HOOK_ACTION_DELETE) != 0) {
+						if ((pact->action & MOM_HOOK_SEND_ACTIONS) != 0) {
+							/* there's a current send action, so delete action should not execute first */
+							pact->do_delete_action_first = 0;
+						} else {
+							/* there's no current send action, so delete action should execute first */
+							pact->do_delete_action_first = 1;
+						}
```

Comment:
This if else block can be shortened to
```
pact->do_delete_action_first = (pact->action & MOM_HOOK_SEND_ACTIONS) ? 1 : 0;
```
or even
```
pact->do_delete_action_first = !!(pact->action & MOM_HOOK_SEND_ACTIONS);
```
But I feel the first version is more readable.